vsintegration/src/FSharp.Editor/Common/Extensions.fs

```diff
@@ -115,6 +115,40 @@ module String =
                 yield String.Empty
         |]
 
+    let getNonEmptyLines (str: string) =
+        use reader = new StringReader(str)
+        [|
+        let line = ref (reader.ReadLine())
+        while not (isNull (!line)) do
+            if (!line).Length > 0 then
+                yield !line
+            line := reader.ReadLine()
+        |]
+
+    let lowerCaseFirstChar (str: string) =
+        if String.IsNullOrEmpty str 
+         || Char.IsLower(str, 0) then str else 
+        let strArr = str.ToCharArray()
+        match Array.tryHead strArr with
+        | None -> str
+        | Some c  -> 
+            strArr.[0] <- Char.ToLower c
+            String (strArr)
+
+
+    let extractTrailingIndex (str: string) =
```

Comment:
Such functions are tricky to write correctly. I'd prefer to have some tests (FsCheck fits the best).